model-optimizer/extensions/ops/interpolate.py

```diff
@@ -69,3 +92,65 @@ def infer(node: Node):
         node.out_port(0).data.set_shape(output_shape)
 
         PermuteAttrs.create_permute_attrs(node, attrs=[('axes', 'input:0')])
+
+    @staticmethod
+    def get_axes(node: Node) -> np.ndarray:
+        opset = node.get_opset()
+        if opset in ['opset1', 'extension']:
+            interp_axes = node.soft_get('axes', None)
+            return interp_axes if interp_axes is None else int64_array(interp_axes)
+
+        src_shape = node.in_port(0).data.get_shape()
+        assert src_shape is not None
+        input_rank = len(src_shape)
+
+        if len(node.in_ports()) == 2:
+            axes = list(range(0, input_rank))
+        else:
+            axes = node.in_port(2).get_source().data.get_value()
+        return int64_array(axes)
+
+    @staticmethod
+    def infer_for_opset4(node: Node):
+        assert len([p for p in node.in_ports().values() if not p.disconnected()]) in [2, 3], \
+            "Interpolate node {} must have 2 or 3 inputs".format(node.soft_get(node.name, node.id))
+        assert node.has_valid('mode')
+        src_shape = node.in_port(0).data.get_shape()
+        assert src_shape is not None
+
+        input_rank = len(src_shape)
+
+        pads_begin = correct_pad(node.soft_get('pads_begin', [0]), input_rank)
```

Comment:
Why do you use this `correct_pad` functions? The `pads_begin` and `pads_end` values must be aligned with the input tensor from the very beginning.

Reply:
It is impossible to implement, because there are front transformations (e.g., transformations `NearestNeighborUpsampling`, `ResizeToInterpolate2D`, `ResizeToInterpolate3D`) generating Interpolate operation, but shapes of inputs are unknown in front stage. Next, in extractors generating Interpolate-4, we do not know input shapes.